compilation stopped working on archlinux due to libxml2 changes

On archlinux with updated packages you now get the following when trying to build (full log):

```
 (master)$ cmake --build . --config Debug
[  1%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/FreetypePrivate.cpp.o
[  1%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/ImageUtils.cpp.o
[  2%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/JpegCommon.cpp.o
[  2%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/PdfDeclarationsPrivate.cpp.o
[  3%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/PdfDrawingOperations.cpp.o
[  4%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/PdfEncodingPrivate.cpp.o
[  4%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/PdfFiltersPrivate.cpp.o
[  5%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/PdfStandard14FontData.cpp.o
[  5%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/PdfStandard14FontFiles.cpp.o
[  6%] Building CXX object src/podofo/private/CMakeFiles/podofo_private.dir/XMPUtils.cpp.o
In file included from /home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:9:
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp: In function ‘xmlNs* findOrCreateNamespace(xmlDocPtr, xmlNodePtr, PdfANamespaceKind)’:
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:19:5: error: ‘xmlErrorPtr’ was not declared in this scope; did you mean ‘xmlDocPtr’?
   19 |     xmlErrorPtr error_ = xmlGetLastError();\
      |     ^~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:207:9: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  207 |         THROW_LIBXML_EXCEPTION(utls::Format("Can't find or create {} namespace", prefix));
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:20:9: error: ‘error_’ was not declared in this scope; did you mean ‘error_t’?
   20 |     if (error_ == nullptr)\
      |         ^~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:207:9: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  207 |         THROW_LIBXML_EXCEPTION(utls::Format("Can't find or create {} namespace", prefix));
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp: In function ‘void addXMPProperty(xmlDocPtr, xmlNodePtr, XMPMetadataKind, PoDoFo::cspan<std::__cxx11::basic_string<char> >&)’:
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:19:5: error: ‘xmlErrorPtr’ was not declared in this scope; did you mean ‘xmlDocPtr’?
   19 |     xmlErrorPtr error_ = xmlGetLastError();\
      |     ^~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:274:9: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  274 |         THROW_LIBXML_EXCEPTION(utls::Format("Can't create xmp:{} node", propName));
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:20:9: error: ‘error_’ was not declared in this scope; did you mean ‘error_t’?
   20 |     if (error_ == nullptr)\
      |         ^~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:274:9: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  274 |         THROW_LIBXML_EXCEPTION(utls::Format("Can't create xmp:{} node", propName));
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp: In function ‘void utls::SetListNodeContent(xmlDocPtr, xmlNodePtr, XMPListType, PoDoFo::cspan<std::__cxx11::basic_string<char> >&, xmlNode*&)’:
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:19:5: error: ‘xmlErrorPtr’ was not declared in this scope; did you mean ‘xmlDocPtr’?
   19 |     xmlErrorPtr error_ = xmlGetLastError();\
      |     ^~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:322:9: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  322 |         THROW_LIBXML_EXCEPTION(utls::Format("Can't create rdf:{} node", elemName));
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:20:9: error: ‘error_’ was not declared in this scope; did you mean ‘error_t’?
   20 |     if (error_ == nullptr)\
      |         ^~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:322:9: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  322 |         THROW_LIBXML_EXCEPTION(utls::Format("Can't create rdf:{} node", elemName));
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:19:5: error: ‘xmlErrorPtr’ was not declared in this scope; did you mean ‘xmlDocPtr’?
   19 |     xmlErrorPtr error_ = xmlGetLastError();\
      |     ^~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:328:13: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  328 |             THROW_LIBXML_EXCEPTION(utls::Format("Can't create rdf:li node"));
      |             ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:20:9: error: ‘error_’ was not declared in this scope; did you mean ‘error_t’?
   20 |     if (error_ == nullptr)\
      |         ^~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:328:13: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  328 |             THROW_LIBXML_EXCEPTION(utls::Format("Can't create rdf:li node"));
      |             ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:19:5: error: ‘xmlErrorPtr’ was not declared in this scope; did you mean ‘xmlDocPtr’?
   19 |     xmlErrorPtr error_ = xmlGetLastError();\
      |     ^~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:337:17: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  337 |                 THROW_LIBXML_EXCEPTION(utls::Format("Can't set xml:lang attribute on rdf:li node"));
      |                 ^~~~~~~~~~~~~~~~~~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XmlUtils.h:20:9: error: ‘error_’ was not declared in this scope; did you mean ‘error_t’?
   20 |     if (error_ == nullptr)\
      |         ^~~~~~
/home/webfreak/dev/podofo/src/podofo/private/XMPUtils.cpp:337:17: note: in expansion of macro ‘THROW_LIBXML_EXCEPTION’
  337 |                 THROW_LIBXML_EXCEPTION(utls::Format("Can't set xml:lang attribute on rdf:li node"));
      |                 ^~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
make[2]: *** [src/podofo/private/CMakeFiles/podofo_private.dir/build.make:202: src/podofo/private/CMakeFiles/podofo_private.dir/XMPUtils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:270: src/podofo/private/CMakeFiles/podofo_private.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

This only started happening right after a system update, which updated libxml2 2.11.5-1 -> 2.12.0-1

Version/git revision: 8bbdacb6d163ccfceb2ef86d134716bdbebdc2ed
Operating System: ArchLinux

--- a/src/podofo/main/PdfXMPPacket.cpp
+++ b/src/podofo/main/PdfXMPPacket.cpp
@@ -8,6 +8,7 @@
 #include "PdfXMPPacket.h"
 #include <podofo/private/XmlUtils.h>
 #include <libxml/xmlsave.h>
+#include <libxml/parser.h>
 #include <podofo/private/XMPUtils.h>
 
 using namespace std;

--- a/src/podofo/private/XmlUtils.h
+++ b/src/podofo/private/XmlUtils.h
@@ -8,6 +8,7 @@
 
 #include <string>
 #include <libxml/tree.h>
+#include <libxml/xmlerror.h>
 #include <podofo/auxiliary/nullable.h>
 
 // Cast macro that keep or enforce const to use
@@ -16,7 +17,7 @@
 
 #define THROW_LIBXML_EXCEPTION(msg)\
 {\
-    xmlErrorPtr error_ = xmlGetLastError();\
+    const xmlError* error_ = xmlGetLastError();\
     if (error_ == nullptr)\
         PODOFO_RAISE_ERROR_INFO(PdfErrorCode::XmpMetadata, msg);\
     else\

